### ISSUE TITLE
GigaPath slide encoder: coordinate normalization collapses all tile positions to a 4×4 grid and missing global_pool=True

### ISSUE THREAD

**Issue by qinghezeng** [KatherLab/STAMP#159]
Hi Duke / STAMP team, thanks for building and maintaining this great framework! I believe I've found two bugs in the GigaPath slide encoder — credit to @xwumed for identifying the issue.

## Summary

`src/stamp/encoding/encoder/gigapath.py` has two critical issues that together render the GigaPath slide-level encoder essentially non-functional:

**Bug 1 — Coordinate normalization destroys spatial information**
`_convert_coords` normalizes tile coordinates to `[0, 1000]` before passing them to the GigaPath LongNet. However, `LongNetViT.coords_to_pos()` in the upstream `prov-gigapath` package internally divides coordinates by `tile_size=256` (pixels) to compute grid indices. The two steps conflict: normalizing to `[0, 1000]` then dividing by 256 collapses all tiles to just ~16 unique positions (4×4 grid) out of 1,000,000 possible, destroying all spatial information. You can verify this by computing pairwise cosine similarity of GigaPath slide embeddings across patients — all slides yield near-identical vectors (~0.98–0.99) instead of the diverse values (~0.4–0.7) seen with working encoders.

**Bug 2 — Untrained CLS token used instead of global pooling**
The slide encoder is created without `global_pool=True`, so the model returns the CLS token. The official GigaPath documentation [explicitly warns](https://github.com/prov-gigapath/prov-gigapath/blob/main/demo/4_load_slide_encoder.py) that the CLS token was **not trained during pretraining**. The official inference pipeline and demos always use `global_pool=True`, which returns the mean pool over all patch tokens — a meaningful, trained representation.

### Bug 1 — Coordinates
The initial commit (`506f60f`, 2025-04-30) added `_convert_coords` with the comment *"For some reason gigapath requires normalized coords in a [0,1000] grid"*. In reality, `LongNetViT.coords_to_pos()` expects **raw pixel coordinates** and performs the grid mapping itself via `floor(coords / tile_size)`. The official GigaPath tiling pipeline stores level-0 pixel coordinates in tile filenames (e.g., `00256x_00512y.png`) and passes them directly to the model.

Since STAMP stores coordinates in **micrometers (µm)**, the correct fix is to convert µm → pixels by dividing by MPP (`coords_um / mpp`), **not** to pass raw µm values directly. Passing µm would still be incorrect because the `tile_size=256` divisor is in pixels — at 20× magnification (MPP=0.5), tiles spaced 256 px apart are only 128 µm apart, so `floor(128 / 256) = 0` and every pair of adjacent tiles would collide into the same grid position, losing ~75% of spatial resolution.

| MPP | Magnification | Spatial resolution loss if µm passed directly |
|-----|---------------|-----------------------------------------------|
| 0.25 µm/px | 40× | ~94% (4 tiles per grid cell) |
| 0.5 µm/px | 20× | ~75% (2 tiles per grid cell) |
| 1.0 µm/px | 10× | ~0% (accidentally correct) |

For reference, STAMP's own Titan encoder handles this correctly:
```python
# From titan.py
coords_px = coords_tensor / coords.mpp   # µm → pixels
```

### Bug 2 — global_pool
`slide_encoder.create_model()` defaults to `global_pool=False`. When `False`, `LongNetViT.forward()` returns the CLS token (`x[:, 0]`). The official demo explicitly notes:

```python
# NOTE: CLS token is not trained during the pretraining
tile_encoder, slide_encoder_model = load_tile_slide_encoder(global_pool=True)
```

Both bugs have been present since the first slide encoder commit and remain unfixed through all v2.4.x releases.

## Suggested fix

Three changes in `stamp/encoding/encoder/gigapath.py`, plus deletion of the now-unused `_convert_coords` method.

### 1. In `__init__`, pass `global_pool=True` when creating the model

Replace:
```python
model = slide_encoder.create_model(
    "hf_hub:prov-gigapath/prov-gigapath", "gigapath_slide_enc12l768d", 1536
)
```
with:
```python
model = slide_encoder.create_model(
    "hf_hub:prov-gigapath/prov-gigapath", "gigapath_slide_enc12l768d", 1536,
    global_pool=True
)
```

### 2. In `_generate_slide_embedding` (lines 54–68), convert µm → pixels

Replace:
```python
# Calculate slide dimensions
slide_width = max(coords.coords_um[:, 0]) + coords.tile_size_um
slide_height = max(coords.coords_um[:, 1]) + coords.tile_size_um

# Normalize coordinates to a [0, 1000] grid
n_grid = 1000
norm_coords = self._convert_coords(
    coords.coords_um, slide_width, slide_height, n_grid, current_x_offset=0
)
norm_coords = (
    torch.tensor(norm_coords, dtype=torch.float32)
    .unsqueeze(0)
    .to(device)
    .half()
)
```
with:
```python
coords_px = coords.coords_um / coords.mpp  # µm → pixel coordinates
norm_coords = (
    torch.tensor(coords_px, dtype=torch.float32)
    .unsqueeze(0)
    .to(device)
    .half()
)
```

### 3. In `encode_patients_` (lines 164–182), convert µm → pixels with offset

Replace:
```python
norm_coords = self._convert_coords(
    coords=coords.coords_um,
    total_wsi_width=total_wsi_width,
    max_wsi_height=max_wsi_height,
    n_grid=1000,
    current_x_offset=current_x_offset,
)

# Update x-coordinates by shifting them based on the current_x_offset
current_x_offset += (
    wsi_width  # Move the x_offset forward for the next slide
)

norm_coords = (
    torch.tensor(norm_coords, dtype=torch.float32)
    .unsqueeze(0)
    .to(device)
    .half()
)
```
with:
```python
offset_coords_um = coords.coords_um.copy()
offset_coords_um[:, 0] += current_x_offset

current_x_offset += wsi_width

coords_px = offset_coords_um / coords.mpp  # µm → pixel coordinates
norm_coords = (
    torch.tensor(coords_px, dtype=torch.float32)
    .unsqueeze(0)
    .to(device)
    .half()
)
```

### 4. Delete `_convert_coords` (lines 215–236)

This method is no longer needed and can be removed entirely.

Thanks for looking into this!
Qinghe
